### PR TITLE
Hacktoberfest: Add solution for Project Euler: Problem 89

### PR DESCRIPTION
**Pull request by coryallegory**
### **Describe your change:**

Adding a Python implementation for Project Euler problem 89
https://projecteuler.net/problem=89

Contributes to Issue #2695 

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by coryallegory** [comment]
`/scripts/validate_filenames.py` appears to fail when including .txt files in the problem directory. I see that Problem 22 also fails if the validation script is run. I'm not sure why it's being triggered as a failure after having a look at the test cases...

**Review by dhruvmanila — COMMENTED** [review]
### Announcement:

@coryallegory

*This message is sent to all pull requests which are **OPEN, NOT MERGED, not containing 'wontfix' label and contains 'Project Euler' or 'Project' or 'Euler' (case doesn't matter) in their title**. If this message doesn't apply to your pull request, please ignore this.*

### Message:

This is to notify all the users submitting a pull request for the Project Euler solution that the [README.md](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md) file has been updated with all the style requirements necessary for the directory and solution files. ***Please read through it and make all the necessary changes required to your submission.***

**Review by 53jk1 — APPROVED** [review]

File: project_euler/problem_089/sol1.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+Find the number of characters saved by writing each of these in their minimal form.
+
+Note: You can assume that all the Roman numerals in the file contain no more than four
+consecutive identical units.
+"""
+
+import os
+
+symbols = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
```
Comment (dhruvmanila) [inline project_euler/problem_089/sol1.py:33]:
```suggestion
SYMBOLS = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
```
This is a global constant

File: project_euler/problem_089/sol1.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    e.g.
+    >>> parse_roman_numerals("LXXXIX")
+    89
+    >>> parse_roman_numerals("IIII")
+    4
+    """
+
+    num = 0
+
+    i = 0
```
Comment (dhruvmanila) [inline project_euler/problem_089/sol1.py:48]: Descriptive names please.

File: project_euler/problem_089/sol1.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    e.g.
+    >>> generate_roman_numerals(89)
+    'LXXXIX'
+    >>> generate_roman_numerals(4)
+    'IV'
+    """
+
+    numerals = ""
+
+    m = num // 1000
```
Comment (dhruvmanila) [inline project_euler/problem_089/sol1.py:74]: If possible can this and other variables defined in this function be descriptive?

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

**Comment by dhruvmanila** [comment]
Awesome work on making another file for testing! 🎉 

There's still the merge conflict. You need to rebase on the latest commit from master.

File: project_euler/problem_089/sol1.py (hunk 2 above)
Comment (coryallegory) [reply project_euler/problem_089/sol1.py:48]: It's a looping index counter, i is pretty much universally used. Not sure what you're expecting in its place.

File: project_euler/problem_089/sol1.py (hunk 3 above)
Comment (coryallegory) [reply project_euler/problem_089/sol1.py:74]: the variables are literally representing the roman numeral letters

**Review by dhruvmanila — APPROVED** [review]
Awesome work! Thanks for doing this :)

**Pull request merged**